common/src/main/java/com/wynntils/handlers/item/ItemHandler.java

```diff
@@ -251,6 +254,32 @@ private ItemAnnotation calculateAnnotation(ItemStack itemStack, StyledText name)
         return annotation;
     }
 
+    private StyledText simplifyName(StyledText name) {
+        String str = name.getString();
+
+        final Pattern[] PATTERNS = {
+            Pattern.compile("^§6(?:Buying|Selling) [^ ]+ (.+?)(?:§6)? for .+ Each$"),
+            Pattern.compile("^§6(?:Buying|Selling) [^ ]+ (.+)$"),
+            Pattern.compile("^§7§l[^ ]+x (.+)$")
+        };
+
+        for (Pattern p : PATTERNS) {
+            Matcher m = p.matcher(str);
+
+            if (m.matches()) {
+                str = m.group(1);
+                break;
+            }
+        }
+
+        // Sometimes the text before the item name is already white, so the server omits this prefix
```

Comment:
This only happens if you match the pattern and extract a partial name, no? So move this into the for...if statement.

Reply:
In fact, you don't need a break there, do the entire  return StyledText.fromString(str), and if you fall through the for loop, just return the original `name`.